core/blockchain.go

```diff
@@ -1279,6 +1323,38 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 	return bc.writeBlockWithState(block, receipts, logs, internalTxs, state, emitHeadEvent)
 }
 
+// reorgNeeded determines if the external chain is better than the local chain so reorg is needed
+func (bc *BlockChain) reorgNeeded(localBlock *types.Block, localTd *big.Int, externBlock *types.Block, externTd *big.Int) bool {
+	if consensusEngine, ok := bc.engine.(consensus.FastFinalityPoSA); ok {
+		localJustifiedBlockNumber, _ := consensusEngine.GetJustifiedBlock(bc, localBlock.NumberU64(), localBlock.Hash())
+		externJustifiedBlockNumber, _ := consensusEngine.GetJustifiedBlock(bc, externBlock.NumberU64(), externBlock.Hash())
+
+		if externJustifiedBlockNumber > localJustifiedBlockNumber {
+			return true
+		}
+	}
+
```

Comment:
do we need to add: if externJustifiedBlockNumber < localJustifiedBlockNumber, return false?

Reply:
Yes, I will fix this